client/index.js

```diff
@@ -65,10 +66,27 @@ async function doRender ({ Component, props, err }) {
     props = await loadGetInitialProps(Component, { err, pathname, query })
   }
 
+  if (Component === ErrorComponent) {
+    lastScroll = {
+      x: window.pageXOffset,
+      y: window.pageYOffset
+    }
+  }
+
   Component = Component || lastAppProps.Component
   props = props || lastAppProps.props
 
   const appProps = { Component, props, err, router, headManager }
-  lastAppProps = appProps
   ReactDOM.render(createElement(App, appProps), container)
+
+  if (lastScroll &&
+    Component !== ErrorComponent &&
+    lastAppProps.Component === ErrorComponent) {
```

Comment:
Yeah I got it.
I mean only this check: `lastAppProps.Component === ErrorComponent`.

Since we also check for `lastScroll`, seems like this is like a always true clause. 
Give it a try to remove it and test. If we need it, keep it like this.